etc/helm/pachyderm/templates/_helpers.tpl

```diff
@@ -24,13 +24,44 @@ LOCAL
 {{- end -}}
 
 {{- define "pachyderm.consoleSecret" -}}
-{{ default (randAlphaNum 32) .Values.console.config.oauthClientSecret }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.console.config.oauthClientSecret "hashSalt" "consoleSecret" "onlyDefault" .Release.IsUpgrade) }}
 {{- end -}}
 
 {{- define "pachyderm.clusterDeploymentId" -}}
-{{ default (randAlphaNum 32) .Values.pachd.clusterDeploymentID }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.pachd.clusterDeploymentID "hashSalt" "deploymentID" "onlyDefault" .Release.IsUpgrade) }}
 {{- end -}}
 
 {{- define "pachyderm.enterpriseSecret" -}}
-{{ default (randAlphaNum 32) .Values.pachd.enterpriseSecret }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.pachd.enterpriseSecret "hashSalt" "enterpriseSecret" "onlyDefault" .Release.IsUpgrade) }}
+{{- end -}}
+
+
+## Input:
+## This function expects a context containing 'defaultValue', 'hashSalt' and 'onlyDefault' keys
+##
+## Behavior: 
+## Use 'defaultValue' if it's provided, otherwise use the current date/time to create a stable hash
+## incorporating the 'hashSalt' value to produce a different hash per client template's usage. 
+## This way different caller templates can produce different random values.
+##
+## Discussions: 
+## 1.) The reason we use a a truncated time string to create the hash, 
+## is because we want to this template to produce the same hash each 
+## time it's executed in a release. In this case, the current timestamp 
+## gives us a unique enough value to use as a hash seed. We truncate it 
+## to the minute to reduce the odds that it will produce different timestamp 
+## values between renderings of this template.
+## Reference https://github.com/helm/helm/issues/6456
+##
+## 2.) 'onlyDefault' is used to force this template to fail if a default value is not provided
+{{- define "defaultOrStableHash" -}}
+{{- if .defaultValue }}
```

Comment:
This method might be a little "too smart" for helm (specifically passing in a hash as an arg) Generally with helm it's ok to break DRY to make things more clear / use values directly.